src/value.h

```diff
@@ -84,6 +86,20 @@ typedef struct RSValue {
       uint8_t staticarray : 1;
     } arrval;
 
+    struct {
+      /**
+       * Duo value
+       * 
+       * Allows keeping a value, together with an addition value.
+       * 
+       * For example, keeping a value and, in addition, a different value for serialization, such as a JSON String representation.
+       */ 
+      
+      // An array of 2 RSValue *'s
+      // The first entry is the value, the second entry is the additional value
+      struct RSValue **vals;
```

Comment:
Why you need a pointer to array of pointers, we can have a pointer to array of `RSValue` and space a deference. Or if you choose to keep it you can use the `arrval` struct no?

Reply:
All usages / callees  are expecting `RSValue*`, so it seems we can keep it as it is now and avoid the copy by value, and the address-of operators.
Regarding `arrval` struct - it cannot be used as-is since a new type with different behavior is needed to differentiate it from none-duo values,
In addition, `arrval` has some data I don't need (I know `duoval` array has only 2 entries and is not static)

Reply:
2 pointers or fixed size array will increase `sizeof`